File: src/main/java/io/curiousoft/ijudi/ordermanagement/validator/MobileNumberValidator.java
package io.curiousoft.ijudi.ordermanagement.validator;

import io.curiousoft.ijudi.ordermanagement.utils.IjudiUtils;

import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;

public class MobileNumberValidator implements ConstraintValidator<ValidMobileNumber, String> {

    @Override
    public boolean isValid(String value, ConstraintValidatorContext context) {
        return IjudiUtils.isSAMobileNumber(value);
    }
}

File: izinga-commons/src/main/kotlin/io/curiousoft/izinga/commons/validator/MobileNumberValidator.kt
package io.curiousoft.izinga.commons.validator

import javax.validation.ConstraintValidator
import javax.validation.ConstraintValidatorContext

class MobileNumberValidator : ConstraintValidator<ValidMobileNumber?, String> {
    override fun isValid(value: String, context: ConstraintValidatorContext): Boolean {
        return isSAMobileNumber(value)
    }

    fun isSAMobileNumber(number: String?): Boolean {
        if (number == null) return false
        val regex = "(\\+27|27|0)[1-9]\\d{8}"
        return number.matches(regex.toRegex())
    }
}